File: week2_0710/week2-additional.js
//Additional Task 1
function maxProduct(nums){
    let maxProduct = nums[0]*nums[1];
    for (let i=0; i<nums.length-1;i++){
        for (let j=i+1;j<nums.length;j++){
            let product= nums[i]*nums[j]
            if(product>maxProduct){
                maxProduct=product
            }
        }
    }
    console.log(maxProduct)
}
console.log("=====Task1=====")
maxProduct([10,-20,0,3])//print 30
maxProduct([10,-20,0,-3])//print 60
maxProduct([5,20,2,6])//print 120
maxProduct([-1,2])//print -2
maxProduct([-1,0,2])//print 0
maxProduct([5,-1,-2,0])//print 2
maxProduct([-5,-2])//print 10


//Additional Task 2
function twoSum(nums,target){
    let ansIndex=[]
    for (let i=0; i<nums.length-1;i++){
        for (let j=i+1;j<nums.length;j++){
            if(nums[i]+nums[j]==target){
                ansIndex.push(i);
                ansIndex.push(j);
                return ansIndex
            }
        }
    }
    if(ansIndex==[]){
        console.log("None")
    }
}
console.log("=====Task2=====")
let result = twoSum([2,11,7,15],9);
console.log(result);

File: week2_0710/week2-additional.py
#Additional Task 1
def max_product(nums):
    maxProduct=nums[0]*nums[1]
    for i in range(len(nums)-1):
        for j in range (i+1,len(nums)):
            product=nums[i]*nums[j] 
            if(product>maxProduct):
                maxProduct=product
    print(maxProduct)            
print("=====Task1=====")
max_product([5,20,2,6])#print 120
max_product([10,-20,0,3])#print 30
max_product([10,-20,0,-3])#print 60
max_product([-1,2])#print -2
max_product([-1,0,2])#print 0
max_product([5,-1,-2,0])#print 2
max_product([-5,-2])#print 10


#Additional Task 2
def two_sum(nums, target):
    ansIndex = []
    for i in range(len(nums)-1):
        for j in range(i+1,len(nums)):
            if((nums[i]+nums[j])==target):
                ansIndex.append(i)
                ansIndex.append(j)
                return ansIndex
    if(ansIndex==[]):
        print("None!")

        
print("=====Task2=====")
result=two_sum([2,11,7,15],9)
print(result)